timesketch/lib/llms/features/llm_forensic_report.py

```diff
@@ -0,0 +1,304 @@
+            FileNotFoundError: If the prompt file cannot be found.
+            IOError: If there's an error reading the prompt file.
+            OSError: If there's an error reading the prompt file.
+        """
+        prompt_file_path = current_app.config.get(self.PROMPT_CONFIG_KEY)
+        if not prompt_file_path:
+            logger.error("%s config not set", self.PROMPT_CONFIG_KEY)
+            raise ValueError("LLM forensic report prompt path not configured.")
+
+        try:
+            with open(prompt_file_path, "r", encoding="utf-8") as file_handle:
+                prompt_template = file_handle.read()
+        except FileNotFoundError as exc:
+            logger.error("Forensic report prompt file not found: %s", prompt_file_path)
+            raise FileNotFoundError(
+                f"LLM Prompt file not found: {prompt_file_path}"
+            ) from exc
+        except IOError as e:
+            logger.error("Error reading prompt file: %s", e)
+            raise IOError("Error reading LLM prompt file.") from e
+
+        if "<EVENTS_JSON>" not in prompt_template:
+            logger.error("Prompt template is missing the <EVENTS_JSON> placeholder")
+            raise ValueError(
+                "LLM forensic report prompt template is missing the "
+                "required <EVENTS_JSON> placeholder."
+            )
+
+        prompt_text = prompt_template.replace("<EVENTS_JSON>", json.dumps(events_dict))
+        return prompt_text
+
+    def _run_timesketch_query(
+        self,
+        sketch: Sketch,
+        query_string: str = "*",
+        query_filter: Optional[dict] = None,
+        id_list: Optional[list] = None,
+        datastore: Optional[OpenSearch] = None,
+        timeline_ids: Optional[list] = None,
+    ) -> pd.DataFrame:
+        """Runs a timesketch query and returns results as a DataFrame.
+        Args:
+            sketch: The Sketch object to query.
+            query_string: Search query string.
+            query_filter: Dictionary with filter parameters.
+            id_list: List of event IDs to retrieve.
+            datastore: OpenSearch instance for querying.
+            timeline_ids: List of timeline IDs to query.
+        Returns:
+            pd.DataFrame: DataFrame containing query results.
+        Raises:
+            ValueError: If datastore is not provided or no valid indices are found.
+        """
+        if datastore is None:
+            raise ValueError("Datastore must be provided.")
+
+        if not query_filter:
+            query_filter = {}
+
+        if id_list:
+            id_query = " OR ".join([f'_id:"{event_id}"' for event_id in id_list])
+            query_string = id_query
+
+        all_indices = list({t.searchindex.index_name for t in sketch.timelines})
+        indices_from_filter = query_filter.get("indices", all_indices)
+
+        if "_all" in indices_from_filter:
+            indices_from_filter = all_indices
```

Comment:
This is actually a logic error (we just fixed it in PR #3359 ). So this should be:

```suggestion
        all_timeline_ids = [t.id for t in sketch.timelines]
        indices = query_filter.get("indices", all_timeline_ids)

        if "_all" in indices_from_filter:
            indices_from_filter = all_timeline_ids
```